session.go

```diff
@@ -335,6 +335,9 @@ func (s *session) run() error {
 
 	var closeErr closeError
 	handshakeEvent := s.handshakeEvent
+	var pacingDeadline time.Time
```

Comment:
Can you make this a member var as all the other deadlines? Then you can drop the param to resetTimer() and simplify some of the code below iiuc.